app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,54 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.order(market_name: :asc)
+    @all_markets = Market.order(market_name: :asc)
+  end
+
+  def show
+    @market = Market.find(params[:id])
```

Comment:
You often will have to save the market (or whatever it is in the controller) based on the params[:id]. If you find this repeated in your controller, it is a good candidate for pulling out into a private method. For example, you could do:
```def get_market
   @market = Market.find(params[:id])
end```
And at the beginning of the controller, you could do a before action to say call that private method before the show and edit methods.

You probably haven't learned about before actions yet, but I thought I'd just give you a heads up!